Assets/MixedRealityToolkit-SDK/Inspectors/UX/Interactable/InspectorBase.cs

```diff
@@ -0,0 +1,646 @@
+                    SerializedProperty textureValue = prop.FindPropertyRelative("TextureValue");
+                    EditorGUILayout.PropertyField(textureValue, new GUIContent(label.stringValue, tooltip.stringValue), false);
+                    break;
+                case InspectorField.FieldTypes.Vector2:
+                    SerializedProperty vector2Value = prop.FindPropertyRelative("Vector2Value");
+                    vector2Value.vector2Value = EditorGUILayout.Vector2Field(new GUIContent(label.stringValue, tooltip.stringValue), vector2Value.vector2Value);
+                    break;
+                case InspectorField.FieldTypes.Vector3:
+                    SerializedProperty vector3Value = prop.FindPropertyRelative("Vector3Value");
+                    vector3Value.vector3Value = EditorGUILayout.Vector3Field(new GUIContent(label.stringValue, tooltip.stringValue), vector3Value.vector3Value);
+                    break;
+                case InspectorField.FieldTypes.Vector4:
+                    SerializedProperty vector4Value = prop.FindPropertyRelative("Vector4Value");
+                    vector4Value.vector4Value = EditorGUILayout.Vector4Field(new GUIContent(label.stringValue, tooltip.stringValue), vector4Value.vector4Value);
+                    break;
+                case InspectorField.FieldTypes.Curve:
+                    SerializedProperty curveValue = prop.FindPropertyRelative("CurveValue");
+                    curveValue.animationCurveValue = EditorGUILayout.CurveField(new GUIContent(label.stringValue, tooltip.stringValue), curveValue.animationCurveValue);
+                    break;
+                case InspectorField.FieldTypes.Quaternion:
+                    SerializedProperty quaternionValue = prop.FindPropertyRelative("QuaternionValue");
+                    Vector4 vect4 = new Vector4(quaternionValue.quaternionValue.x, quaternionValue.quaternionValue.y, quaternionValue.quaternionValue.z, quaternionValue.quaternionValue.w);
+                    vect4 = EditorGUILayout.Vector4Field(new GUIContent(label.stringValue, tooltip.stringValue), vect4);
+                    quaternionValue.quaternionValue = new Quaternion(vect4.x, vect4.y, vect4.z, vect4.w);
+                    break;
+                case InspectorField.FieldTypes.AudioClip:
+                    SerializedProperty audioClip = prop.FindPropertyRelative("AudioClipValue");
+                    EditorGUILayout.PropertyField(audioClip, new GUIContent(label.stringValue, tooltip.stringValue), false);
+                    break;
+                case InspectorField.FieldTypes.Event:
+                    SerializedProperty uEvent = prop.FindPropertyRelative("EventValue");
+                    if (uEvent != null)
+                    {
+                        //Debug.Log("update? " + uEvent);
+                    }
+                    EditorGUILayout.PropertyField(uEvent, new GUIContent(label.stringValue, tooltip.stringValue));
```

Comment:
We should probably display differently if we can't edit events. At the very least, there should be a label that they're readonly.

Reply:
Events are a weird situation, there was talk on an earlier PR about removing UnityEvents and replace them with a pattern being used in other features of MRTK. I haven't got around to looking at that yet. I can definitely clean this up for now.

Reply:
I'd really like to do that I'd we can. Hopefully I'll get some time to do a deep dive and open a PR with any suggestions there.

Reply:
I removed the comments in this section of code. Should I resolve this because there are UnityEvent discussions going on elsewhere?

Reply:
The only pattern I've seen is more of a callback system so developers could extend a class and handle events in their specific code. That's definitely doable and a preferred method for doing this in code. There is another thread about using UnityEvents in the inspector only so designers could wire things up if they need to, which is super important because a lot of this is for designers so they can create nice buttons, but they have to be able to test them as well without code. I'll start looking into that. This code is only about what shows up in the inspector, so I will dig into the developer side soon.